logzio-sender/src/test/java/io/logz/sender/DiskQueueTest.java

```diff
@@ -0,0 +1,116 @@
+                                              ScheduledExecutorService tasks,
+                                              boolean compressRequests) throws LogzioParameterErrorException {
+
+        LogzioTestStatusReporter logy = new LogzioTestStatusReporter(logger);
+        HttpsRequestConfiguration httpsRequestConfiguration = HttpsRequestConfiguration
+                .builder()
+                .setCompressRequests(compressRequests)
+                .setConnectTimeout(serverTimeout)
+                .setSocketTimeout(socketTimeout)
+                .setLogzioToken(token)
+                .setLogzioType(type)
+                .setLogzioListenerUrl("http://" + getMockListenerHost() + ":" + getMockListenerPort())
+                .build();
+
+        if (bufferDir == null) {
+            bufferDir = TestEnvironment.createTempDirectory();
+            bufferDir.deleteOnExit();
+        }
+
+        int fsPercentThreshold = zeroThresholdBuffer ? 0 : FS_PERCENT_THRESHOLD;
+
+        LogzioSender logzioSender = LogzioSender
+                .builder()
+                .setDebug(false)
+                .setTasksExecutor(tasks)
+                .setDrainTimeout(drainTimeout)
+                .setReporter(logy)
+                .setHttpsRequestConfiguration(httpsRequestConfiguration)
+                .WithDiskMemoryQueue()
+                    .setBufferDir(bufferDir)
+                    .setFsPercentThreshold(fsPercentThreshold)
+                    .setCheckDiskSpaceInterval(1000)
+                .EndDiskQueue()
+                .build();
+
+        logzioSender.start();
+        return logzioSender;
+    }
+
+    @Override
+    protected void setZeroThresholdBuffer() throws LogzioParameterErrorException {
+        zeroThresholdBuffer = true;
+    }
+
+    private void setBufferDir(File bufferDir) {
+        this.bufferDir = bufferDir;
+    }
+
+    @Test
+    public void testLoggerCantWriteToEmptyDirectory() {
+        String token = "nowWeTestLoggerCantWriteToTmpDirectory";
+        String type = "justTestingNoWriteDir";
+        String loggerName = "changeBufferLocation";
+        int drainTimeout = 10;
+        File tempDirectory = new File("" + File.separator);
+        String message1 = "Just sending something - " + random(5);
+        ScheduledExecutorService tasks = Executors.newScheduledThreadPool(3);
+        try {
+            setBufferDir(tempDirectory);
+            LogzioSender testSender = createLogzioSender(token, type, drainTimeout, 10 * 1000,
+                    10 * 1000, tasks, false);
+        } catch(LogzioParameterErrorException e) {
+            assertTrue(e.getMessage().contains(tempDirectory.getAbsolutePath()));
+        }
+        assertTrue(tempDirectory.exists());
+        tempDirectory.delete();
+        tasks.shutdownNow();
+    }
+
+    @Test
+    public void testLoggerCreatesDirectoryWhichDoesNotExists() throws Exception {
+        String token = "nowWeWantToChangeTheBufferLocation";
```

Comment:
You have too much boilerplate, reporting it self, making very hard to focus on what actually being tested here.